protocol/messenger_handler.go

```diff
@@ -1574,21 +1574,44 @@ func (m *Messenger) HandleDeleteMessage(state *ReceivedMessageState, deleteMessa
 		}
 	}
 
+	var messagesToDelete []*common.Message
+	// In case of Image messages, we need to delete all the images in the album
+	if originalMessage.ContentType == protobuf.ChatMessage_IMAGE {
+		image := originalMessage.GetImage()
+		messagesInTheAlbum, err := m.persistence.albumMessages(deleteMessage.ChatId, image.GetAlbumId())
+		if err != nil {
+			return err
+		}
+		for _, messageInAlbum := range messagesInTheAlbum {
+			messagesToDelete = append(messagesToDelete, messageInAlbum)
+		}
+	} else {
+		messagesToDelete = append(messagesToDelete, originalMessage)
+	}
+
 	// Update message and return it
-	originalMessage.Deleted = true
-	originalMessage.DeletedBy = deleteMessage.DeleteMessage.DeletedBy
+	for _, messageToDelete := range messagesToDelete {
```

Comment:
@jrainville, do you know if the action of deleting messages is happening inside a transaction? If not, then although rare, it would be better if we could save all messages in one go (query) or do one by one and rollback in case one of them fail before the loop finishes.

Reply:
I don't think it happens in a transaction.
As for rolling back, I'm not sure if it's necessary. Since the goal is to delete, if one of the messages fails, it's ok if the some of them are deleted IMO.
What would happen on the front is that one of the images would disappear, the user could try to delete again for the remaining ones.

Reply:
Thanks for taking a look @jrainville. It would be a bit better if we could be more consistent by default, but as you said and I agree, in the rare case that something happens the user could just try again, so it's fine.